src/components/dls/ContentModal/ContentModal.tsx

```diff
@@ -53,10 +56,12 @@ const ContentModal = ({
   hasHeader = true,
   onClick,
   shouldBeFullScreen = false,
+  isOverlayMax = false,
+  isBottomSheetOnMobile,
 }: ContentModalProps) => {
   const overlayRef = useRef<HTMLDivElement>();
   const { locale } = useRouter();
-
+  const isMobile = useIsMobile();
```

Comment:
Consider using CSS media queries instead of detecting mobile via JavaScript to apply conditional class names. Media queries are more performant, reliable, and follow best practices for responsive design. They avoid layout shifts, reduce JS complexity, and work better across edge cases like resized desktop windows or tablets. Use JS only when detecting capabilities CSS can’t handle (e.g., touch support).